devel/ci/integration/tests/utils.py

```diff
@@ -54,6 +57,32 @@ def make_db_and_user(db_container, name, use_dump=False):
         db_container.execute(["rm", "/tmp/database.dump"])
 
 
+def wait_for_file(
+        container: conu.DockerContainer, path: str, dir_not_empty: bool = False,
+        timeout: int = 120):
+    """Check that a file is in a container.
+
+    Args:
+        container: The container where the file is.
+        path: The file path in the container.
+        timeout: How long to wait before throwing an exception.
+    """
+    while timeout > 0:
```

Comment:
I don't think this can ever be `False` with `if timeout == 0:" in the loop below. You could just move the latter check (and raise the exception) out of the loop:

```
while timeout > 0:
    ...
    timeout -= 1
if timeout == 0:
    raise ...
```

Reply:
Yep, agreed.